src/wallet/wallet.cpp

```diff
@@ -3831,7 +3831,7 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
         return nullptr;
     }
 
-    if (!gArgs.GetArg("-changetype", "").empty() && !ParseOutputType(gArgs.GetArg("-changetype", ""), walletInstance->m_default_change_type)) {
+    if (!gArgs.GetArg("-changetype", "").empty() && !ParseOutputType(gArgs.GetArg("-changetype", ""), *(walletInstance->m_default_change_type = OutputType{}))) {
```

Comment:
I'm not a fan of this assignment inside an `if` statement. See more verbose suggestion below.

Reply:
`ParseOutputType` assigns a new value to the argument, so why can't the argument itself not be initialized?

Note that with C++17, we can use `walletInstance->m_default_change_type.emplace()` instead. 

```suggestion
    if (!gArgs.GetArg("-changetype", "").empty() && !ParseOutputType(gArgs.GetArg("-changetype", ""), *walletInstance->m_default_change_type.emplace())) {
```

I am happy to change it to that once we switch to C++17

Reply:
I don't think it's bad, but the line is hard to read. Another approach could be to initialise `m_default_change_type` in `wallet.h`; that doesn't seem unreasonable for an Optional.

Reply:
Agree that this is a bit hard to read.

Reply:
thx changed